src-java/floodlight-service/floodlight-modules/src/main/java/org/openkilda/floodlight/command/rulemanager/SwitchDataProvider.java

```diff
@@ -0,0 +1,117 @@
+/* Copyright 2021 Telstra Open Source
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package org.openkilda.floodlight.command.rulemanager;
+
+import org.openkilda.floodlight.converter.rulemanager.OfFlowConverter;
+import org.openkilda.floodlight.converter.rulemanager.OfGroupConverter;
+import org.openkilda.floodlight.converter.rulemanager.OfMeterConverter;
+import org.openkilda.messaging.MessageContext;
+import org.openkilda.model.SwitchFeature;
+import org.openkilda.model.SwitchId;
+import org.openkilda.rulemanager.FlowSpeakerData;
+import org.openkilda.rulemanager.GroupSpeakerData;
+import org.openkilda.rulemanager.MeterSpeakerData;
+
+import lombok.Builder;
+import lombok.extern.slf4j.Slf4j;
+import net.floodlightcontroller.core.IOFSwitch;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+
+@Slf4j
+public class SwitchDataProvider {
```

Comment:
Please, provide a javadoc for the class.

Reply:
Added.